### ISSUE TITLE
Implementation does not work for numbers larger than PHP_INT_MAX

### ISSUE THREAD

**Issue by rtek** [Ekman/Luhn-Algorithm#9]
This implementation assumes that that the "number" is an integer when it should be a string of [0-9].

Consider SIM card ICCID numbers which are greater than PHP_INT_MAX 64bit.

```
    public function testOne(): void
    {
        $luhn = new LuhnAlgorithm();
        $this->assertTrue($luhn->isValid(Number::fromString('89148000003974165685')));
    }
Failed asserting that false is true.

    public function testTwo(): void
    {
        $number = Number::fromString($input = '89148000003974165685');
        $this->assertEquals($input, $number->getNumber());
    }

Failed asserting that 922337203685477632 matches expected '89148000003974165685'.
```

**Comment by laurent35240** [comment]
Hi,

I run into same issue when I wanted to validate a number with 19 digits.
We should not use int type for handling such numbers

**Comment by Ekman** [comment]
Thanks for reporting this! I've fixed the issue in #10, which will be merged and tagged shortly.
